.github/workflows/wheels.yml

```diff
@@ -59,48 +42,78 @@ jobs:
       CIBW_ARCHS_MACOS: "x86_64 arm64 universal2"
       CIBW_ENVIRONMENT_MACOS: "MACOSX_DEPLOYMENT_TARGET=10.13 CMAKE_OSX_ARCHITECTURES=\"${{ matrix.os.cibw-arch == 'macosx_x86_64' && 'x86_64' || matrix.os.cibw-arch == 'macosx_arm64' && 'arm64' || matrix.os.cibw-arch == 'macosx_universal2' && 'arm64;x86_64' || '' }}\""
 
+    strategy:
+      fail-fast: false
+      matrix:
+        # Launch separate job for each python. The build is so much longer than
+        # machine configuration/setup, so parallel builds will be faster. More
+        # importantly, github times out after 6 hours _per job_.
+        cpversion:
+          - cp36
+          - cp37
+          - cp38
+          - cp39
+          - cp310
+          - cp311
+          - cp312
+        os:
+          - runs-on: ubuntu-latest
+            cibw-arch: manylinux_x86_64
+          - runs-on: macos-latest
+            cibw-arch: macosx_x86_64
+          - runs-on: macos-latest
+            cibw-arch: macosx_arm64
+          - runs-on: windows-latest
+            cibw-arch: win_amd64
+        exclude:
+          - cpversion: cp36
+            os:
+              runs-on: macos-latest
+              cibw-arch: macosx_arm64
+          - cpversion: cp37
+            os:
+              runs-on: macos-latest
+              cibw-arch: macosx_arm64
 
     steps:
-      - uses: actions/checkout@v3
-        with:
-          submodules: 'recursive'
+      - uses: actions/checkout@v4
 
-      - uses: actions/setup-python@v2
+      - uses: actions/setup-python@v5
         name: Install Python
-        with:
-          python-version: '3.7'
 
       - name: Install cibuildwheel
         run: |
           python -m pip install --upgrade pip
-          python -m pip install cibuildwheel==2.12.0
+          python -m pip install cibuildwheel
 
       - name: Build wheels
         run: |
           python -m cibuildwheel --output-dir wheelhouse
 
       # Upload binaries to github
-      - uses: actions/upload-artifact@v2
+      - uses: actions/upload-artifact@v4
         with:
-          path: |
+          name: artifact-wheel-${{ matrix.cpversion }}-${{ matrix.os.cibw-arch }}
```

Comment:
Required due to `v4` change https://github.com/actions/upload-artifact/blob/4c0ff1c489dca52fedb26375d7d8fe7bd9233f19/docs/MIGRATION.md?plain=1#L74-L80